pkg/artifactory/resource_xray_watch.go

```diff
@@ -164,199 +111,12 @@ func resourceXrayWatch() *schema.Resource {
-	resource := new(WatchProjectResource)
-
-	cfg := rawCfg.(map[string]interface{})
-	resource.Type = StringPtr(cfg["type"].(string))
-	if v, ok := cfg["bin_mgr_id"]; ok {
-		resource.BinaryManagerId = StringPtr(v.(string))
-	}
-	if v, ok := cfg["repo_type"]; ok {
-		resource.RepoType = StringPtr(v.(string))
-	}
-	if v, ok := cfg["name"]; ok {
-		resource.Name = StringPtr(v.(string))
-	}
-	if v, ok := cfg["filters"]; ok {
-		resourceFilters := expandFilters(v.([]interface{}))
-		resource.Filters = &resourceFilters
-	}
-
-	return resource
-}
-
-func expandFilters(l []interface{}) []WatchFilter {
-	filters := make([]WatchFilter, 0, len(l))
-
-	for _, raw := range l {
-		filter := new(WatchFilter)
-		f := raw.(map[string]interface{})
-		filter.Type = StringPtr(f["type"].(string))
-		valueWrapper := new(WatchFilterValueWrapper)
-		fv := new(WatchFilterValue)
-		fv.Value = StringPtr(f["value"].(string))
-		valueWrapper.WatchFilterValue = *fv
-		filter.Value = valueWrapper
-
-		filters = append(filters, *filter)
-	}
-
-	return filters
-}
-
-func expandAssignedPolicy(rawCfg interface{}) *WatchAssignedPolicy {
-	policy := new(WatchAssignedPolicy)
-
-	cfg := rawCfg.(map[string]interface{})
-	policy.Name = StringPtr(cfg["name"].(string))
-	policy.Type = StringPtr(cfg["type"].(string))
-
-	return policy
-}
-
-func flattenProjectResources(resources *WatchProjectResources) []interface{} {
-	if resources == nil || resources.Resources == nil {
-		return []interface{}{}
-	}
-
-	var l []interface{}
-	for _, res := range *resources.Resources {
-		m := make(map[string]interface{})
-		m["type"] = res.Type
-		if res.Name != nil {
-			m["name"] = res.Name
-		}
-		if res.BinaryManagerId != nil {
-			m["bin_mgr_id"] = res.BinaryManagerId
-		}
-		if res.RepoType != nil {
-			m["repo_type"] = res.RepoType
-		}
-		m["filters"] = flattenFilters(res.Filters)
-		l = append(l, m)
-	}
-
-	return l
-}
-
-func flattenFilters(filters *[]WatchFilter) []interface{} {
-	if filters == nil {
-		return []interface{}{}
-	}
-
-	var l []interface{}
-	for _, f := range *filters {
-		m := make(map[string]interface{})
-		m["type"] = f.Type
-		m["value"] = f.Value.WatchFilterValue.Value
-		l = append(l, m)
-	}
-
-	return l
-}
-
-func flattenAssignedPolicies(policies *[]WatchAssignedPolicy) []interface{} {
-	if policies == nil {
-		return []interface{}{}
-	}
-
-	var l []interface{}
-	for _, p := range *policies {
-		m := make(map[string]interface{})
-		m["name"] = p.Name
-		m["type"] = p.Type
-		l = append(l, m)
-	}
-
-	return l
-}
-
 func resourceXrayWatchCreate(d *schema.ResourceData, m interface{}) error {
-
-	watch := expandWatch(d)
-	_, err := m.(*resty.Client).R().SetBody(watch).Post("xray/api/v2/watches")
-	if err != nil {
-		return err
-	}
-
-	d.SetId(*watch.GeneralData.Name) // ID may be returned according to the API docs, but not in go-xray
-	return resourceXrayWatchRead(d, m)
+	return errors.New("Use Xray provider resource instead")
```

Comment:
This is nice of you, but I would literally remove everything.

Reply:
This is from TF guideline.

Reply:
Can you link we the guide?

Reply:
https://www.terraform.io/plugin/sdkv2/best-practices/deprecations#provider-data-source-or-resource-removal